src/main/java/ch/njol/skript/config/Config.java

```diff
@@ -168,17 +173,14 @@ public String getFileName() {
 	/**
 	 * Saves the config to a file.
 	 *
-	 * @param f The file to save to
+	 * @param file The file to save to
 	 * @throws IOException If the file could not be written to.
 	 */
-	public void save(final File f) throws IOException {
-		separator = defaultSeparator;
-		final PrintWriter w = new PrintWriter(f, "UTF-8");
-		try {
-			main.save(w);
-		} finally {
-			w.flush();
-			w.close();
+	public void save(File file) throws IOException {
+		this.separator = defaultSeparator;
+		try (final PrintWriter writer = new PrintWriter(file, StandardCharsets.UTF_8)) {
+			this.main.save(writer);
+			writer.flush();
```

Comment:
does this need to go in a `finally`?

Reply:
No, flushing should be fine after the write. If an IO error was thrown during the writing then there'd be nothing to flush anyway. (The error would either be thrown when opening the writer, or when actually flushing the buffer to the file so it would probably be this line that causes the IO error if any!)